File: backend/ŠišAppApi/Models/Salon.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace ŠišAppApi.Models;

public class Salon
{
    [Key]
    public int Id { get; set; }
    
    [Required]
    [MaxLength(100)]
    public string Name { get; set; } = string.Empty;
    
    public string? Description { get; set; }
    
    [Required]
    [MaxLength(255)]
    public string Address { get; set; } = string.Empty;
    
    [Required]
    [MaxLength(50)]
    public string City { get; set; } = string.Empty;
    
    [Required]
    [MaxLength(20)]
    public string PostalCode { get; set; } = string.Empty;
    
    [Required]
    [MaxLength(50)]
    public string Country { get; set; } = string.Empty;
    
    public double? Latitude { get; set; }
    
    public double? Longitude { get; set; }
    
    [Required]
    [MaxLength(20)]
    public string Phone { get; set; } = string.Empty;
    
    [MaxLength(100)]
    [EmailAddress]
    public string? Email { get; set; }
    
    [MaxLength(255)]
    public string? Website { get; set; }
    
    public string? ImageIds { get; set; } // JSON array of image IDs
    
    [Required]
    [Range(0, 5)]
    public double Rating { get; set; } = 0;
    
    public int ReviewCount { get; set; } = 0;
    
    public bool IsVerified { get; set; } = false;
    
    public DateTime? VerifiedAt { get; set; }
    
    public string? VerificationNotes { get; set; }
    
    public string? BusinessHours { get; set; } // JSON object with business hours
    
    public string? Amenities { get; set; } // JSON array of amenities
    
    public string? SocialMedia { get; set; } // JSON object with social media links
    
    public bool IsActive { get; set; } = true;
    
    [Required]
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    
    public DateTime? UpdatedAt { get; set; }
    
    public bool IsDeleted { get; set; } = false;
    
    public DateTime? DeletedAt { get; set; }
    
    // Navigation properties
    public virtual ICollection<Barber> Barbers { get; set; } = new List<Barber>();
    public virtual ICollection<Service> Services { get; set; } = new List<Service>();
    public virtual ICollection<Appointment> Appointments { get; set; } = new List<Appointment>();
    public virtual ICollection<Review> Reviews { get; set; } = new List<Review>();
} 

File: sisapp_frontend/lib/models/salon.dart
class Salon {
  final int id;
  final String name;
  final int cityId;
  final String city;
  final String address;
  final String phone;
  final int employeeCount;
  final double rating;

  final String postalCode;
  final String? website;
  final String? imageIds;
  final double? latitude;
  final double? longitude;
  final List<String>? services;
  bool isActive;

  Salon({
    required this.id,
    required this.name,
    required this.cityId,
    required this.city,
    required this.address,
    required this.phone,
    required this.postalCode,
    this.website,
    this.imageIds,
    this.latitude,
    this.longitude,
    this.services,
    this.employeeCount = 0,
    this.rating = 0.0,
    this.isActive = true,
  });

  factory Salon.fromJson(Map<String, dynamic> json) {
    return Salon(
      id: (json['id'] as num?)?.toInt() ?? 0,
      name: json['name'] ?? '',
      cityId: (json['cityId'] as num?)?.toInt() ?? 0,
      city: json['city'] ?? '',
      address: json['address'] ?? '',
      phone: json['phone'] ?? '',
      postalCode: json['postalCode'] ?? '',
      website: json['website'],
      imageIds: json['imageIds'],
      employeeCount: (json['employeeCount'] as num?)?.toInt() ?? 0,
      rating: (json['rating'] as num?)?.toDouble() ?? 0.0,
      latitude: (json['latitude'] as num?)?.toDouble(),
      longitude: (json['longitude'] as num?)?.toDouble(),
      services: json['services'] != null ? List<String>.from(json['services']) : null,
      isActive: json['isActive'] ?? true,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'cityId': cityId,
      'city': city,
      'address': address,
      'phone': phone,
      'postalCode': postalCode,
      'website': website,
      'imageIds': imageIds,

      'rating': rating,
      'latitude': latitude,
      'longitude': longitude,
      'services': services,
      'employeeCount': employeeCount,
      'isActive': isActive,
    };
  }

  @override
  bool operator ==(Object other) {
    if (identical(this, other)) return true;
    return other is Salon && other.id == id;
  }

  @override
  int get hashCode => id.hashCode;
}
